src/main/java/com/microsoft/azure/vmagent/AzureVMAgent.java

```diff
@@ -526,6 +550,118 @@ public synchronized void deprovision(Localizable reason) throws Exception {
                 new Object[]{this.getDisplayName(), reason == null ? "Unknown reason" : reason.toString()});
 
         computer.setAcceptingTasks(false);
+
+        ComputerLauncher launcher = computer.getLauncher();
+
+        if ((launcher instanceof AzureVMAgentSSHLauncher)) {
+            AzureVMAgentSSHLauncher azureLauncher = (AzureVMAgentSSHLauncher) launcher;
+
+            OutputStream terminateResults = new OutputStream() {
```

Comment:
I am still confused about this stream, why not use the log stream as the init script does?

Reply:
For reference you are referring to the local variable 'logger' on line 103 of AzureVMAgentSSHLauncher?  The logger reference used here is pulled with the line: PrintStream logger = listener.getLogger();  The listener object here is passed as a parameter to the launch function which executes the init script code and the deprovision function does not provide any such parameter for me to use for logging purposes.  

Therefore the best design I found was to use the implementation above since the AzureVMAgent does have access to the LOGGER variable which can at least provide general feedback to the system logs for the deprovisioning process and script execution.  If there is an alternative I would definitely be open to trying it, but I didn't see another way to accomplish the logging without changing some large pieces of object access and/or Jenkins extensibility points.

Reply:
One possible solution is `PrintStream terminateStream = new LogTaskListener(LOGGER, Level.INFO).getLogger();`.

Reply:
That appears to work pretty well, I was not aware of that object in hudson model.